syncd/Syncd.cpp

```diff
@@ -5296,3 +5296,23 @@ syncd_restart_type_t Syncd::handleRestartQuery(
 
     return RequestShutdownCommandLineOptions::stringToRestartType(op);
 }
+
+bool Syncd::shouldDiscoverPortObjects() const
+{
+    SWSS_LOG_ENTER();
+
+#ifdef SKIP_SAI_PORT_DISCOVERY_ON_FAST_BOOT
+    const bool discoverPortObjectsInFastBoot = false;
+#else
+    const bool discoverPortObjectsInFastBoot = true;
+#endif
```

Comment:
fast boot cak be initiated after code was compiled which then this check will be hardcoded

Reply:
also there are no tests for testing this code

Reply:
> fast boot cak be initiated after code was compiled which then this check will be hardcoded

This was the intention

Reply:
this should be runtime check

Reply:
@kcudnik What is the benefit of runtime check here? Syncd is compiled per platform and on Nvidia we do not want to run discovery. We know this at compile time.